server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java

```diff
@@ -316,30 +336,82 @@ public void handleResult(TaskResult taskResult) {
             log.warn("Job is finished, ignore result, jobId={}, currentStatus={}", je.getId(), je.getStatus());
             return;
         }
-        int rows = updateJobScheduleEntity(taskResult, je);
+        // TODO: update task entity only when progress changed
+        int rows = updateJobEntity(taskResult, je);
         if (rows > 0) {
             taskResultPublisherExecutor
                     .execute(() -> publisher.publishEvent(new DefaultJobProcessUpdateEvent(taskResult)));
+
             if (publisher != null && taskResult.getStatus() != null && taskResult.getStatus().isTerminated()) {
                 taskResultPublisherExecutor.execute(() -> publisher
                         .publishEvent(new JobTerminateEvent(taskResult.getJobIdentity(), taskResult.getStatus())));
+
                 if (taskResult.getStatus() == JobStatus.FAILED) {
                     AlarmUtils.alarm(AlarmEventNames.TASK_EXECUTION_FAILED,
                             MessageFormat.format("Job execution failed, jobId={0}",
                                     taskResult.getJobIdentity().getId()));
                 }
             }
         }
-
     }
 
-    @Transactional(rollbackFor = Exception.class)
     @Override
-    public void handleHeart(HeartRequest heart) {
-        if (heart.getJobIdentity() == null || heart.getJobIdentity().getId() == null ||
-                heart.getExecutorEndpoint() == null) {
+    public void refreshResult(Long id) {
+        taskResultPullerExecutor.execute(() -> {
+            try {
+                doRefreshResult(id);
+            } catch (Exception e) {
+                log.error("Refresh job result failed, jobId={}", id, e);
+            }
+        });
+    }
+
+    private void doRefreshResult(Long id) throws JobException {
```

Comment:
the operation logic is same no matter push mode and pull mode, why shouldn't we call handleResult when pull mode, just like what we do in push mode~

Reply:
Play to refactor later, current sprint, we keep minimum change for PUSH monitor mode